### PR TITLE
Add standard helm templating and linting ci

### LINKED ISSUE #110: refactor: use standard Helm templating
**Describe the Feature**

Currently, the UI uses direct Zarf variable replacement within its charts instead of the standard Helm values replacement pattern. Refactor the existing charts to take on a Helm templating pattern with a values file that isolates the Zarf ENV variable replacement instead.

**Additional context**

See [leapfrogai-model-skeleton](https://github.com/defenseunicorns/leapfrog-model-skeleton/tree/main/chart) or [leapfrogai-api](https://github.com/defenseunicorns/leapfrogai-api/tree/main/chart) for examples.

### PR DESCRIPTION
**Pull request by lemonprogis**
Moved the Zarf ENV variables to values.yaml and adjusted the templates accordingly. Fixes #110 

In addition, worked with @justinthelaw to integrate some of the developer workflow enhancements from his PR #137.

### CODE REVIEW

**Comment by justinthelaw** [comment]
@lemonprogis Thanks for helping us out and contributing! Several things:

1. The workflow version needs to be the SHA to a commit related to a tagged release
2. The workflow doesn't seem to work and may be pointing at a non-existent workflow or version
3. Did you test the new helm charts using a helm hydration dry-run?
4. Did you test the helm charts with a Zarf deployment to a cluster?

File: .github/workflows/lint-chart.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      - name: Checkout Repo
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          fetch-depth: 0
+
+      - name: Set up Helm
+        uses: azure/setup-helm@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
```
Comment (justinthelaw) [inline .github/workflows/lint-chart.yml:18]: Needs SHA to commit

File: .github/workflows/lint-chart.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        with:
+          python-version: 3.12
+
+      - name: Set up chart-testing
+        uses: helm/chart-testing-action@v3.10.1
```
Comment (justinthelaw) [inline .github/workflows/lint-chart.yml:23]: Needs SHA to commit, also runner was not able to pull this.

**Review by justinthelaw — COMMENTED** [review]
Needs evidence of local Zarf deployment test

**Comment by lemonprogis** [comment]
> @lemonprogis Thanks for helping us out and contributing! Several things:
> 
> 1. The workflow version needs to be the SHA to a commit related to a tagged release
> 2. The workflow doesn't seem to work and may be pointing at a non-existent workflow or version
> 3. Did you test the new helm charts using a helm hydration dry-run?
> 4. Did you test the helm charts with a Zarf deployment to a cluster?
> 
> EDIT: If you aren't sure how to use Zarf or spin-up a test cluster, please let me know. You can get started with https://zarf.dev/

Thanks, @justinthelaw. I appreciate the feedback! I'll test this against zarf and commit any relevant changes. Also, would it make sense to lint the helm chart in the 'release' workflow instead (or should linting happen in a git hook)? I had that thought last night and wanted to check with y'all.

**Comment by justinthelaw** [comment]
@lemonprogis I am actually currently implementing leapfrogai, organization wide changes to our repos, to include pre-commit hooks, dev processes and CI/check pipelines, similar to the helm chart workflow that you have here. 

Ideally, the workflows mirror the pre-commit hooks. An example pre-commit config is below. If you could add a helm chart one and demonstrate its capability, that'd be great!

```yaml
repos:
    - repo: https://github.com/pre-commit/pre-commit-hooks
      rev: v4.0.1
      hooks:
          - id: check-added-large-files
            name: Large Files check
            args: ['--maxkb=1024']

          - id: check-merge-conflict
            name: Merge Conflict Resolution Check

          - id: end-of-file-fixer
            name: Newline EOF Checker

          - id: fix-byte-order-marker
            name: Fix UTF-8 byte order marker

          - id: trailing-whitespace
            name: Whitespace Cleaning Check
            args: [--markdown-linebreak-ext=md]

    - repo: https://github.com/gitleaks/gitleaks
      rev: v8.18.0
      hooks:
      - id: gitleaks
        name: GitLeaks Checks

    - repo: https://github.com/sirosen/fix-smartquotes
      rev: 0.2.0
      hooks:
          - id: fix-smartquotes

    - repo: https://github.com/dnephin/pre-commit-golang
      rev: v0.4.0
      hooks:
          - id: go-fmt
            name: Golang Formatting

    - repo: https://github.com/astral-sh/ruff-pre-commit
      rev: v0.2.1
      hooks:
          - id: ruff
            name: Ruff Linting
            args: [--fix]
            types_or: [python, pyi, jupyter]
            pass_filenames: false
          - id: ruff-format
            name: Ruff Formatting
            types_or: [python, pyi, jupyter]
            pass_filenames: false

    - repo: https://github.com/pre-commit/mirrors-eslint
      rev: v8.56.0
      hooks:
          - id: eslint
            name: ESLint and Prettier
            files: \.([jt]sx?|svelte)$ # *.js, *.jsx, *.ts, *.tsx and *.svelte
            types: [file]
            additional_dependencies:
                - eslint@8.44.0
                - eslint-plugin-prettier@5.0.0
                - eslint-plugin-svelte@2.35.1
            args: [--fix]

    - repo: https://github.com/pryorda/dockerfilelint-precommit-hooks
      rev: v0.1.0
      hooks:
          - id: dockerfilelint
            name: 'Dockerfile Lint'
            stages: [commit]

    # get latest zarf schema
    - repo: local
      hooks:
          - id: download-schema
            name: 'Download Zarf Schema'
            entry: bash -c 'curl -o zarf.schema.json https://raw.githubusercontent.com/defenseunicorns/zarf/main/zarf.schema.json'
            language: system

    # check against latest zarf schema
    - repo: https://github.com/python-jsonschema/check-jsonschema
      rev: 0.14.0
      hooks:
          - id: check-jsonschema
            name: 'Validate Zarf Configs Against Schema'
            files: 'zarf.yaml'
            types: [yaml]
            args: ['--schemafile', 'zarf.schema.json']

    # clean-up latest zarf schema
    - repo: local
      hooks:
          - id: delete-schema
            name: 'Delete Zarf Schema'
            entry: bash -c 'rm -f zarf.schema.json'
            language: system


    # get latest uds schema
    - repo: local
      hooks:
          - id: download-schema
            name: 'Download UDS Schema'
            entry: bash -c 'curl -o uds.schema.json https://raw.githubusercontent.com/defenseunicorns/uds-cli/main/uds.schema.json'
            language: system

    # check against latest uds schema
    - repo: https://github.com/python-jsonschema/check-jsonschema
      rev: 0.14.0
      hooks:
          - id: check-jsonschema
            name: 'Validate UDS Configs Against Schema'
            files: 'uds.yaml'
            types: [yaml]
            args: ['--schemafile', 'uds.schema.json']

    # clean-up latest uds schema
    - repo: local
      hooks:
          - id: delete-schema
            name: 'Delete UDS Schema'
            entry: bash -c 'rm -f uds.schema.json'
            language: system

    # naive approaches for checking dependencies versus project definitions

    - repo: local
      hooks:
          - id: check-paired-files-python
            name: Check PyProject
            entry: bash -c '(if git diff --cached --name-only | grep -q -E "^(pyproject\.toml)$"; then git diff --cached --name-only | grep -q -E "^(requirements\.txt)$" || echo "Warning, you may need to check your Python dependencies"; fi; if git diff --cached --name-only | grep -q -E "^(requirements\.txt)$"; then git diff --cached --name-only | grep -q -E "^(pyproject\.toml)$" || echo "Warning, you may need to check your Python dependencies"; fi)' --
            language: system
            always_run: true
            pass_filenames: false

    - repo: local
      hooks:
          - id: check-paired-files-node
            name: Check NPM Packages
            entry: bash -c '(if git diff --cached --name-only | grep -q -E "^(package\.json)$"; then git diff --cached --name-only | grep -q -E "^(package-lock\.json)$" || echo "Warning, you may need to check your NPM dependencies"; fi; if git diff --cached --name-only | grep -q -E "^(package-lock\.json)$"; then git diff --cached --name-only | grep -q -E "^(package\.json)$" || echo "Warning, you may need to check your NPM dependencies"; fi)' --
            language: system
            always_run: true
            pass_filenames: false
```

**Comment by lemonprogis** [comment]
> @lemonprogis Thanks for helping us out and contributing! Several things:
> 
> 1. The workflow version needs to be the SHA to a commit related to a tagged release
> 2. The workflow doesn't seem to work and may be pointing at a non-existent workflow or version
> 3. Did you test the new helm charts using a helm hydration dry-run?
> 4. Did you test the helm charts with a Zarf deployment to a cluster?
> 
> EDIT: If you aren't sure how to use Zarf or spin-up a test cluster, please let me know. You can get started with https://zarf.dev/

Hey Justin! I 'think' I got a test cluster going and have deployed the ui using zarf, I did run into an error where the UDS Package CRD was there. I installed that next along with adding those to my k3d cluster. I deployed again and got further along, but the pod never spun up under the deployment. I passed it the version 0.3.0 to pull that image. I can share logs etc, but going to try a fresh setup today and work on those commit hooks.

File: .github/workflows/lint-chart.yml (hunk 2 above)
Comment (lemonprogis) [reply .github/workflows/lint-chart.yml]: removed this since we'll go with pre commit hooks

File: .github/workflows/lint-chart.yml (hunk 1 above)
Comment (lemonprogis) [reply .github/workflows/lint-chart.yml]: removed since we'll go with pre commit hookse

**Comment by lemonprogis** [comment]
> @lemonprogis Thanks for helping us out and contributing! Several things:
> 
> 1. The workflow version needs to be the SHA to a commit related to a tagged release
> 2. The workflow doesn't seem to work and may be pointing at a non-existent workflow or version
> 3. Did you test the new helm charts using a helm hydration dry-run?
> 4. Did you test the helm charts with a Zarf deployment to a cluster?
> 
> EDIT: If you aren't sure how to use Zarf or spin-up a test cluster, please let me know. You can get started with https://zarf.dev/

my local needs more horse power apparently :D 

image


but it's there 

image

**Comment by lemonprogis** [comment]
@justinthelaw, I added pre-commit linting for the helm chart.

**Comment by justinthelaw** [comment]
Coverage is being performed via overarching and superseding PR: #137. Thank you for the input and conversation!

**Pull request closed without merging**